gax-grpc/src/main/java/com/google/api/gax/grpc/GrpcDirectStreamController.java

```diff
@@ -99,6 +100,35 @@ void start(RequestT request) {
     }
   }
 
+  ApiStreamObserver<RequestT> startBidiStreaming() {
+    responseObserver.onStart(this);
+    this.hasStarted = true;
+
+    clientCall.start(new ResponseObserverAdapter(), new Metadata());
+    if (autoflowControl) {
+      clientCall.request(1);
+    } else if (numRequested > 0) {
+      clientCall.request(numRequested);
+    }
+
+    return new ApiStreamObserver<RequestT>() {
+      @Override
+      public void onCompleted() {
+        clientCall.halfClose();
+      }
+
+      @Override
+      public void onError(Throwable t) {
+        clientCall.cancel("user sent error", t);
+      }
+
+      @Override
+      public void onNext(RequestT request) {
+        clientCall.sendMessage(request);
+      }
+    };
```

Comment:
Might be good to add some state sanity checks here: calling any method after onCompleted & onError should be an error